fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -411,6 +411,49 @@ class BackupContainerFileSystemImpl {
 		}
 	}
 
+	// Checks if list of sorted logfiles have the logs from snapshotBeginVersion to snapshotEndversion.
+	// Which means the sorted log files(have beginVersion and endVersion) should cover
+	// all the versions between snapshotBegingVersion and snapshotEndversion.
+	// Note: logs should be sorted
+	static bool hasContinuousLogsForSnapshot(const std::vector<LogFile>& logs,
+	                                         Version snapshotBeginVersion,
+	                                         Version snapshotEndVersion) {
+		auto it = logs.begin();
+
+		// find the first mutation log file that covers snapshotBeginVersion
+		while (it != logs.end()) {
+			if (it->beginVersion <= snapshotBeginVersion && it->endVersion > snapshotBeginVersion)
+				break;
+			++it;
+		}
+
+		// no log find found covering snaphostBeginVersion, return false
+		if (it == logs.end())
+			return false;
+
+		// If current log entry(it), covers the entire snapshot, return true
+		if (it->endVersion >= snapshotEndVersion)
+			return true;
+
+		// Iterate over the next logs, check if they are continuous and if
+		// the log file is covering the snapshot.
+		Version prevEnd = it->endVersion;
+		++it;
+
+		while (it != logs.end()) {
+			if (it->beginVersion == prevEnd &&
+			    it->endVersion >= snapshotEndVersion) // continuous logs until snapshot is covered
+				return true;
+			else if (it->beginVersion != prevEnd) // not continuous logs
+				return false;
+
+			prevEnd = it->endVersion;
+			++it;
+		} // comes out if the logs are not found until snapshotEndVersion.
+
+		return prevEnd >= snapshotEndVersion;
```

Comment:
Nits: I feel the final return is not reachable because the two returns in while loop covers all cases?

Reply:
If we have an entry covering the snaphostBeginVersion but not including snaphostEndVersion, and not more logs after this. I assume it will go to return.

And also if the while loop ends out of logs without including snaphostEndVersion, it will go to return.

In either case, it just returns false.. But additional check may be useful to confirm things.